### PR TITLE
Reorganise documentation structure

### LINKED ISSUE #25: Rearrange the docs to separate general approach and workflow specific details

### PR DESCRIPTION
**Pull request by tskir**
This is done for better clarity, especially for potential outside readers, and for making adding new demonstrator pipelines easier.

Closes #25.

### CODE REVIEW

File: 4.3-pipelines/environments/tesk.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 This document contains instructions on how to run federated analysis workflows. The protocol for running them is the same; the only difference is the actual Nextflow workflow file to be run, and its parameters. Currently, there are two workflows available:
 * A simple “Hello world” to test your setup. This is available in file [`hello_world.nf`](hello_world.nf).
```
Comment (lvarin) [inline 4.3-pipelines/environments/tesk.md]: hello-world is a dead link, it should be `./4.3-pipelines/demonstrators/hello-world/hello_world.nf`

**Comment by lvarin** [comment]
Other than the broken link, it looks good to me.

**Comment by tskir** [comment]
Thanks for the review @lvarin. I can't seem to be able to locate which link is broken, could you say which one please?

File: 4.3-pipelines/environments/tesk.md (hunk 1 above)
Comment (tskir) [reply 4.3-pipelines/environments/tesk.md]: You're right! Thanks for pointing this out. Now fixed, & merging this PR

**Comment by lvarin** [comment]
:+1:

**Pull request merged**